### ISSUE TITLE
[BUG] I get an 500 when trying to view audit logs from the admin console

### ISSUE THREAD

**Issue by RudolfVonKrugstein** [paperless-ngx/paperless-ngx#7649]
### Description

This Error only occurs if I have created a document and make some changes to it, like creating a note.
When I go on the admin console and open the audit logs: (http://server/admin/auditlog/logentry/) I get a page, where it says an 500 (Internal Server error) occured.

### Steps to reproduce

1. Set `PAPERLESS_AUDIT_LOG_ENABLED: "true"` in your `docker-compose.yml`
2. run `docker-compose up`
3. Log into paperless, and go to `../admin/auditlog/logentry` - note that it works.
4. Create a document.
5. Do some change to the document, like creating a note.
6. Log into paperless, and go to `../admin/auditlog/logentry`. - I now get an 500.

### Webserver logs

```bash
webserver_1  |     fields = ", ".join(changes.keys())
webserver_1  |                        ^^^^^^^^^^^^
webserver_1  | AttributeError: 'str' object has no attribute 'keys'
webserver_1  | [2024-09-07 20:57:50,371] [ERROR] [django.request] Internal Server Error: /admin/auditlog/logentry/
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/db/models/options.py", line 681, in get_field
webserver_1  |     return self.fields_map[field_name]
webserver_1  |            ~~~~~~~~~~~~~~~^^^^^^^^^^^^
webserver_1  | KeyError: 'msg_short'
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/utils.py", line 272, in lookup_field
webserver_1  |     f = _get_non_gfk_field(opts, name)
webserver_1  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/utils.py", line 303, in _get_non_gfk_fie
webserver_1  |     field = opts.get_field(name)
webserver_1  |             ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/db/models/options.py", line 683, in get_field
webserver_1  |     raise FieldDoesNotExist(
webserver_1  | django.core.exceptions.FieldDoesNotExist: LogEntry has no field named 'msg_short'
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
webserver_1  |     raise exc_info[1]
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 42, in inner
webserver_1  |     response = await get_response(request)
webserver_1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
webserver_1  |     raise exc_info[1]
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 284, in _get_response_asy
webserver_1  |     response = await sync_to_async(
webserver_1  |                ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 468, in __call__
webserver_1  |     ret = await asyncio.shield(exec_coro)
webserver_1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/asgiref/current_thread_executor.py", line 40, in run
webserver_1  |     result = self.fn(*self.args, **self.kwargs)
webserver_1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 522, in thread_handler
webserver_1  |     return func(*args, **kwargs)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/response.py", line 114, in render
webserver_1  |     self.content = self.rendered_content
webserver_1  |                    ^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/response.py", line 92, in rendered_content
webserver_1  |     return template.render(context, self._request)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/backends/django.py", line 61, in render
webserver_1  |     return self.template.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 175, in render
webserver_1  |     return self._render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 167, in _render
webserver_1  |     return self.nodelist.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
webserver_1  |     return self.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/loader_tags.py", line 157, in render
webserver_1  |     return compiled_parent._render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 167, in _render
webserver_1  |     return self.nodelist.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
webserver_1  |     return self.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/loader_tags.py", line 157, in render
webserver_1  |     return compiled_parent._render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 167, in _render
webserver_1  |     return self.nodelist.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
webserver_1  |     return self.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/loader_tags.py", line 63, in render
webserver_1  |     result = block.nodelist.render(context)
webserver_1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
webserver_1  |     return self.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/loader_tags.py", line 63, in render
webserver_1  |     result = block.nodelist.render(context)
webserver_1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
webserver_1  |     return SafeString("".join([node.render_annotated(context) for node in self]))
webserver_1  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
webserver_1  |     return self.render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/templatetags/base.py", line 45, in rende
webserver_1  |     return super().render(context)
webserver_1  |            ^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/template/library.py", line 258, in render
webserver_1  |     _dict = self.func(*resolved_args, **resolved_kwargs)
webserver_1  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/templatetags/admin_list.py", line 336, in result_list
webserver_1  |     "results": list(results(cl)),
webserver_1  |                ^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/templatetags/admin_list.py", line 312, in results
webserver_1  |     yield ResultList(None, items_for_result(cl, res, None))
webserver_1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/templatetags/admin_list.py", line 303, in __init__
webserver_1  |     super().__init__(*items)
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/templatetags/admin_list.py", line 213, in items_for_result
webserver_1  |     f, attr, value = lookup_field(field_name, result, cl.model_admin)
webserver_1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/django/contrib/admin/utils.py", line 281, in lookup_field
webserver_1  |     value = attr(obj)
webserver_1  |             ^^^^^^^^^
webserver_1  |   File "/usr/local/lib/python3.11/site-packages/auditlog/mixins.py", line 63, in msg_short
webserver_1  |     fields = ", ".join(changes.keys())
webserver_1  |                        ^^^^^^^^^^^^
webserver_1  | AttributeError: 'str' object has no attribute 'keys'
```

### Browser logs

```bash
GET http://localhost:8000/admin/auditlog/logentry/ 500 (Internal Server Error)
```

### Paperless-ngx version

Paperless-ngx 2.11.6

### Host OS

WSL, Ubunto 22.04

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.11.6",
    "server_os": "Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 1081101176832,
        "available": 954411864064
    },
    "database": {
        "type": "sqlite",
        "url": "/usr/src/paperless/data/db.sqlite3",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0001_initial_squashed_0009_mailrule_assign_tags",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "index_status": "OK",
        "index_last_modified": "2024-09-07T20:49:32.259936Z",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2024-09-07T21:05:00.500908Z",
        "classifier_error": null
    }
}
```

### Browser

chrome

### Configuration changes

PAPERLESS_AUDIT_LOG_ENABLED: "true", PAPERLESS_OCR_LANGUAGE: deu

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by stumpylog** [comment]
Seems like a duplicate of #7482 which was resolved

**Comment by RudolfVonKrugstein** [comment]
Could it be, that in https://github.com/paperless-ngx/paperless-ngx/blob/dev/src/documents/views.py#L681 and https://github.com/paperless-ngx/paperless-ngx/blob/dev/src/documents/views.py#L718 the `json.dumps` have to removed, similar to the fix of #7482?

I Havent looked enough in paperlex-ngx yet to have a dev environment for it, but might look into it.

**Comment by shamoon** [comment]
Yea, missed those thanks. And unfortunately the entries will need to be removed or fixed manually. I suppose it would be good to write a migration to find and fix them.
